ion-schema/src/isl/isl_constraint.rs

```diff
@@ -730,6 +741,29 @@ impl IslConstraintImpl {
                     }
                 }
             }
+            "ieee754_float" => {
+                if value.is_null() {
+                    return invalid_schema_error(
+                        "expected a symbol for an `ieee754_float` constraint, found null",
+                    );
+                }
```

Comment:
This is technically unnecessary because of the `as_symbol()` on L757, right? If I'm reading it correctly, it looks like the reason it's here is to provide a null-specific error message, but we could change it to eliminate this check.

Reply:
Do you mind if I do this in a separate PR, there will be many places where I need to change this.

Reply:
No need to increase the scope of this PR. Just fix this one instance that is part of the PR and create an issue to clean up other instances.

Reply:
Sure. Will do. Thanks!